tfe/data_source_workspace_outputs.go

```diff
@@ -0,0 +1,289 @@
+package tfe
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/zclconf/go-cty/cty"
+	ctyjson "github.com/zclconf/go-cty/cty/json"
+)
+
+const sensitiveValue = "<sensitive>"
+
+type dataSourceWorkspaceOutputs struct {
+	provider *pluginProviderServer
+}
+
+var stderr *os.File
+
+func init() {
+	// There is a issue that occurs when the plugin-go Serve function is used that
+	// causes os.Stderr to be overwritten. There is a fix being worked on for this.
+	stderr = os.Stderr
+}
+
+func newDataSourceWorkspaceOutputs(p *pluginProviderServer) tfprotov5.DataSourceServer {
+	return dataSourceWorkspaceOutputs{
+		provider: p,
+	}
+}
+
+func (d dataSourceWorkspaceOutputs) ReadDataSource(ctx context.Context, req *tfprotov5.ReadDataSourceRequest) (*tfprotov5.ReadDataSourceResponse, error) {
+	resp := &tfprotov5.ReadDataSourceResponse{
+		Diagnostics: []*tfprotov5.Diagnostic{},
+	}
+	client, err := getClient(d.provider.meta.hostname, d.provider.meta.token, false)
```

Comment:
I sort of wonder if there's a way to re-use the same client from the main provider instance rather than setting up a separate client for this half?

I dislike the separate clients, it seems like it would remove quite a bit of redundancy in reparsing those provider attributes. Not critical if we can't do it, but do you see a way, somehow passing it to ` tfe.PluginProviderServer` maybe?

Reply:
Hmm, good point. I think theres a way to do it

Reply:
I changed this up so I pass the client directly to the provider.